### PR TITLE
#555 Add nightly release of a snapshot

### PR DESCRIPTION
**Pull request by SylwiaBrant**
GitHub Action that fires at 2:00 am UTC to build a snapshot image of the current repo and push it into DockerHub.

### CODE REVIEW

**Review by prryb — APPROVED** [review]

**Review by KorusMateusz — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -227,10 +227,10 @@ container-runtime-push: ## Push the container
 	$(CONTAINER_RUNTIME_COMMAND) push $(DOCKER_ORGANIZATION)/$(DOCKER_REGISTRY):$(BUILD_TAG) $(CONTAINER_RUNTIME_EXTRA_ARGS)
 
 .PHONY: container-runtime-snapshot-push
-container-runtime-snapshot-push:
+container-runtime-snapshot-push: container-runtime-build
 	@echo "+ $@"
-	$(CONTAINER_RUNTIME_COMMAND) tag $(DOCKER_REGISTRY):$(GITCOMMIT) $(DOCKER_ORGANIZATION)/$(DOCKER_REGISTRY):$(GITCOMMIT) $(CONTAINER_RUNTIME_EXTRA_ARGS)
-	$(CONTAINER_RUNTIME_COMMAND) push $(DOCKER_ORGANIZATION)/$(DOCKER_REGISTRY):$(GITCOMMIT) $(CONTAINER_RUNTIME_EXTRA_ARGS)
+	$(CONTAINER_RUNTIME_COMMAND) tag $(DOCKER_REGISTRY):$(GITCOMMIT) $(DOCKER_ORGANIZATION)/$(DOCKER_REGISTRY):nightly-$(TAG) $(CONTAINER_RUNTIME_EXTRA_ARGS)
+	$(CONTAINER_RUNTIME_COMMAND) push $(DOCKER_ORGANIZATION)/$(DOCKER_REGISTRY):nightly-$(TAG) $(CONTAINER_RUNTIME_EXTRA_ARGS)
```
Comment (tomaszsek) [inline Makefile:233]: revert

File: .github/workflows/deploy-nightly.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+          make helm-e2e E2E_TEST_ARGS='-ginkgo.v'
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_HUB_USERNAME }}
+          password: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
+
+      - name: Prepare image tag
+        run: echo "SHORTER_SHA=$(git rev-parse --short HEAD)" >> $GITHUB_ENV
```
Comment (tomaszsek) [inline .github/workflows/deploy-nightly.yaml:58]: remove

File: .github/workflows/deploy-nightly.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+      - name: Release Container Runtime
+        run: |
+          make container-runtime-snapshot-push TAG=${{ env.SHORTER_SHA }}
```
Comment (tomaszsek) [inline .github/workflows/deploy-nightly.yaml:62]:
```suggestion
          make container-runtime-snapshot-push
```

File: README.md [hunk 4]
```diff
@@ -66,6 +66,12 @@ Selected content:
 We have a dedicated channel called `#jenkins-operator` on [virtuslab-oss.slack.com](https://virtuslab-oss.slack.com)
 Fill out ([Invite form](https://forms.gle/X3X8qA1XMirdBuEH7)) and come say hi !
 
+## Snapshots between releases
+
+We are trying our best to resolve issues quickly, but they have to wait to be released. If you can't wait for an official 
+docker image release and acknowledge the risk, you can use our unofficial images, which are built nightly. 
+Look for the images with tag "nightly-{hash}", where {hash} is the hash of the master commit that interests you.
```
Comment (tomaszsek) [inline README.md:73]:
```suggestion
Look for the images with tag "{git-hash}", where {hash} is the hash of the master commit that interests you.
```

**Review by tomaszsek — CHANGES_REQUESTED** [review]

**Review by tomaszsek — APPROVED** [review]

**Pull request merged**